presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixInitializer.java

```diff
@@ -160,6 +161,10 @@ public RubixInitializer(
 
     void initializeRubix()
     {
+        if ((!nodeManager.getCurrentNode().isCoordinator() || startServerOnCoordinator) && cacheLocation == null) {
```

Comment:
extract `nodeManager.getCurrentNode().isCoordinator() && !startServerOnCoordinator` as variable `cacheDataOnNode` and use it in to `if`s below.